src/mlpack/tests/main_tests/dbscan_test.cpp

```diff
@@ -0,0 +1,517 @@
+
+  SetInputParam("input", inputData);
+  SetInputParam("epsilon", (double) 1.0);
+
+  mlpackMain();
+
+  arma::Row<size_t> output1;
+  output1 = std::move(CLI::GetParam<arma::Row<size_t>>("assignments"));
+
+  bindings::tests::CleanMemory();
+
+  CLI::GetSingleton().Parameters()["input"].wasPassed = false;
+  CLI::GetSingleton().Parameters()["epsilon"].wasPassed = false;
+
+  SetInputParam("input", inputData);
+  SetInputParam("epsilon", (double) 0.5);
+
+  mlpackMain();
+
+  arma::Row<size_t> output2;
+  output2 = std::move(CLI::GetParam<arma::Row<size_t>>("assignments"));
+
+  BOOST_REQUIRE_GT(arma::accu(output1 != output2), 1);
+}
+
+/**
+ * Check that the cluster assignment is different for different
+ * values of Min Size.
+ */
+BOOST_AUTO_TEST_CASE(DBSCANDiffMinSizeTest)
+{
+  arma::mat inputData;
+  if (!data::Load("iris.csv", inputData))
+    BOOST_FAIL("Unable to load dataset iris.csv!");
+
+  SetInputParam("input", inputData);
+  SetInputParam("epsilon", (double) 0.4);
+  SetInputParam("min_size", (int) 5);
+
+  mlpackMain();
+
+  arma::Row<size_t> output1;
+  output1 = std::move(CLI::GetParam<arma::Row<size_t>>("assignments"));
+
+  bindings::tests::CleanMemory();
+
+  CLI::GetSingleton().Parameters()["input"].wasPassed = false;
+  CLI::GetSingleton().Parameters()["epsilon"].wasPassed = false;
+  CLI::GetSingleton().Parameters()["min_size"].wasPassed = false;
+
+  SetInputParam("input", inputData);
+  SetInputParam("epsilon", (double) 0.5);
+  SetInputParam("min_size", (int) 40);
+
+  mlpackMain();
+
+  arma::Row<size_t> output2;
+  output2 = std::move(CLI::GetParam<arma::Row<size_t>>("assignments"));
+
+  BOOST_REQUIRE_GT(arma::accu(output1 != output2), 1);
+}
+
+/**
+ * Check that number of noise points increases as
+ * the value of epsilon decreases.
+ */
+BOOST_AUTO_TEST_CASE(DBSCANEpsilonNoisePointsTest)
+{
+  arma::mat inputData;
+  if (!data::Load("iris.csv", inputData))
+    BOOST_FAIL("Unable to load dataset iris.csv!");
+
+  SetInputParam("input", inputData);
+  SetInputParam("epsilon", (double) 1.0);
+
+  mlpackMain();
+
+  arma::Row<size_t> output1;
+  output1 = std::move(CLI::GetParam<arma::Row<size_t>>("assignments"));
+
+  int noisePointsOutput1 = 0;
+
+  for (size_t i = 0; i < output1.n_elem; ++i)
+  {
+    if (arma::accu(output1[i] > inputData.n_cols))
+    {
+      noisePointsOutput1++;
+    }
+  }
+
+  bindings::tests::CleanMemory();
+
+  CLI::GetSingleton().Parameters()["input"].wasPassed = false;
+  CLI::GetSingleton().Parameters()["epsilon"].wasPassed = false;
+
+  SetInputParam("input", inputData);
+  SetInputParam("epsilon", (double) 0.4);
+
+  mlpackMain();
+
+  arma::Row<size_t> output2;
+  output2 = std::move(CLI::GetParam<arma::Row<size_t>>("assignments"));
+
+  int noisePointsOutput2 = 0;
+
+  for (size_t i = 0; i < output2.n_elem; ++i)
+  {
+    if (arma::accu(output2[i] > inputData.n_cols))
+    {
+      noisePointsOutput2++;
+    }
+  }
+
+  BOOST_REQUIRE_LE(noisePointsOutput1, noisePointsOutput2);
```

Comment:
I am not sure this is the best strategy for testing here.  What we want to see is that our results are different as `epsilon` changes, but as we change `epsilon` we are not guaranteed that the number of points that are noise change (in the worst case, the assignments stay the same).  I would instead suggest sweeping the value of `epsilon` and `min_points` and making sure the resulting cluster assignments are different for at least one element; this lets us know that the values of `epsilon` and `min_points` are being taken into account by the binding.  So I think the other two tests you have written are fine for that case.

Reply:
No, I had different test for different results if we change the value of epsilon. 
With this test, I wanted to check if noise points increases or in worst case remain the same.
I'll remove the test if it is not required

Reply:
Yep, sounds good.  This should already be tested in `DBSCANTest` (`src/mlpack/tests/dbscan_test.cpp`).  Keep in mind, the purpose of testing the bindings is just to make sure that the options being given get to the `DBSCAN` class correctly.  We have already tested the `DBSCAN` method itself in the other tests.  At the same time, if you see that there are some things about the `DBSCAN` class itself that are not being tested, then definitely feel free to also add tests there. :)

Reply:
I added as many tests as I could think of.. 
Yes, these tests are not required here! I'll remove them
Am I missing any other parameter tests?